### ISSUE TITLE
InputNumber doesn't properly execute the declared onchange event handler

### ISSUE THREAD

**Issue by kito99** [primefaces/primefaces#1012]
Currently, InputNumer wraps the onchange listener specified by the user like so:

``` javascript
        // also copy values on onchange - see #293
        // backup onchange, should be executed after our onchange
        var originalOnchange = this.inputExternal.attr('onchange');
        this.inputExternal.attr('onchange', null);
        this.inputExternal.off('change').on('change', function(event) {
            $this.copyValueToHiddenInput();

            if (originalOnchange) {
                eval(originalOnchange);
            }
        });
```

This however, fails in cases where the onchange listener tries to a return a value (which is common):

``` html
<p:inputNumber ... onchange="return onChange(this, event)"/>
```

I have been able to solve this by making the following change:

``` javascript
        var originalOnchange = this.inputExternal.attr('onchange');
        var onChangeFunc = null;
        if (originalOnchange) {
            onChangeFunc = new Function("event", "$this", "$this.copyValueToHiddenInput(); " + originalOnchange);
        }        
        this.inputExternal.attr('onchange', null);
        this.inputExternal.off('change').on('change', function(event) {
            if (onChangeFunc) {
                return onChangeFunc.call(this, event, $this);
            }
            else {
                $this.copyValueToHiddenInput(); 
            }
        });
```

The drawback to this approach is that `copyValueToHiddenInput()` is still called if the return value is false. It does, however, seem to wrap the onchange handler better.

**Comment by tandraschko** [comment]
So your problem actually is, that you can't skip the "onchange" via return false?

**Comment by kito99** [comment]
Yep.

**Comment by tandraschko** [comment]
@kito99 Why do you need this?
The problem is that onchange is often a AJAX event and therefore the value is the old value when the ajax request fires. No idea why does it work currently but i will revert this change to fix the order.

**Comment by kito99** [comment]
@tandraschko, I don't quite understand. Did this ever get fixed?

**Comment by tandraschko** [comment]
Yep, its fixed in 6.0. I just changed some implementation details but it should still work.
